packages/commonwealth/server/workers/commonwealthConsumer/policies/chainEventCreated/handleLaunchpadTrade.ts

```diff
@@ -0,0 +1,68 @@
+import { events, LaunchpadTrade } from '@hicommonwealth/core';
+import { models } from '@hicommonwealth/model';
+import { BigNumber } from 'ethers';
+import Web3 from 'web3';
+import { z } from 'zod';
+
+export async function handleLaunchpadTrade(
+  event: z.infer<typeof events.ChainEventCreated>,
+) {
+  const {
+    0: traderAddress,
+    1: tokenAddress,
+    2: isBuy,
+    3: communityTokenAmount,
+    4: ethAmount,
+    // 5: protocolEthAmount,
+    6: floatingSupply,
+  } = event.parsedArgs as z.infer<typeof LaunchpadTrade>;
+
+  const token = await models.Token.findOne({
```

Comment:
TBH I am fine with this, but I thought we agreed we would only use models in command/queries

Reply:
This code predates that conversation and we also didn't have the "system actor" concept at that point so creating a command wasn't possible. I would like to keep it like this so I can get this merged ASAP to unblock launchpad. I have to refactor the ChainEventCreated policy/pattern in the next 1.5 months anyway so I can handle applying the patterns we discussed during that refactor.

Reply:
In the next round let's consider if this is a projector instead of a policy->command... similar patterns but diff in the core model since launchpad trades are projections not aggregates.